Python/thread_pthread.h

```diff
@@ -302,6 +308,26 @@ PyThread_get_thread_ident(void)
     return (unsigned long) threadid;
 }
 
+#ifdef PY_HAVE_THREAD_NATIVE_ID
+unsigned long
+PyThread_get_thread_native_id(void)
+{
+    if (!initialized)
+        PyThread_init_thread();
+#ifdef __APPLE__
+    uint64_t native_id;
+    pthread_threadid_np(NULL, &native_id);
```

Comment:
In general, the function *can* fail. But in our case, it cannot fail. So I suggest to add (void) to express that we ignore the return value:

```
/* this call cannot fail: ignore return value */ 
(void)pthread_threadid_np(NULL, &native_id);
```

Implementation of pthread_threadid_np in darwin libpthread, simplified code:

```
int
pthread_threadid_np(pthread_t thread, uint64_t *thread_id)
{
	int res = 0;
	...
	if (thread == NULL) {
		*thread_id = self->thread_id;
	} else {
		...
	}
	return res;
}
```

Full code:
https://github.com/apple/darwin-libpthread/blob/master/src/pthread.c#L1111